### ISSUE TITLE
postgresql plugin crashes using numerical password

### ISSUE THREAD

**Issue by sandrobonazzola** [sosreport/sos#433]
posgresql plugin uses 
  os.environ["PGPASSWORD"] = self.getOption("password")

which is internally converted in:
  os.putenv("PGPASSWORD", self.getOption("password"))

and in the process the string is converted in integer, so if you use a password like 123456 it will fail.
Example for reproducing the issue:

``` bash
# sosreport -o postgresql -k postgresql.password=123456 -k postgresql.username=test -kpostgresql.dbname=test  -kpostgresql.dbhost=localhost

sosreport (version 3.2)

plugin vdsm does not install, skipping: No module named config
This command will collect system configuration and diagnostic
information from this Fedora system. An archive containing the collected
information will be generated in /var/tmp.

For more information on the Fedora Project visit:

  https://fedoraproject.org/

The generated archive may contain data considered sensitive and its
content should be reviewed by the originating organization before being
passed to any third party.

No changes will be made to system configuration.

Press ENTER to continue, or CTRL-C to quit.

Please enter your first initial and last name [mididell.home]: 
Please enter the case id that you are generating this report for: 

 Setting up archive ...
 Setting up plugins ...
postgresql
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/sos/sosreport.py", line 1117, in setup
    plug.setup()
  File "/usr/lib/python2.7/site-packages/sos/plugins/postgresql.py", line 118, in setup
    super(RedHatPostgreSQL, self).setup()
  File "/usr/lib/python2.7/site-packages/sos/plugins/postgresql.py", line 87, in setup
    self.pg_dump()
  File "/usr/lib/python2.7/site-packages/sos/plugins/postgresql.py", line 54, in pg_dump
    os.environ["PGPASSWORD"] = self.get_option("password")
  File "/usr/lib64/python2.7/os.py", line 471, in __setitem__
    putenv(key, item)
TypeError: must be string, not int

 Running plugins. Please wait ...

  Running 1/1: postgresql...        

Creating compressed archive...

Your sosreport has been generated and saved in:
  /var/tmp/sosreport-mididell.home-20141114133202.tar.xz

The checksum is: ccbdf2c2ffe23e35bada101136de11af

Please send this file to your support representative.
```

**Comment by pmoravec** [comment]
Won't something like:

diff --git a/sos/plugins/__init__.py b/sos/plugins/__init__.py
index 4fd85be..ea456d0 100644
--- a/sos/plugins/__init__.py
+++ b/sos/plugins/__init__.py
@@ -361,6 +361,8 @@ class Plugin(object):
         '''set the named option to value.'''
         for name, parms in zip(self.opt_names, self.opt_parms):
             if name == optionname:
+               if type(parms['enabled']) != type(value):
+                   value = type(parms['enabled']) (value)
                 parms['enabled'] = value
                 return True
         else:

be the general fix? It simply tries to re-cast the user-provided option to the type of the default value.

Currently this would break many plugins that provide default value of some option in "wrong" type. E.g. postgresql plugin having:

    option_list = [
..
        ('password', 'password for pg_dump' + password_warn_text, '', False),
..
    ]

while expecting password option to be string. (s/False/'' fixes the plugin, roughly speaking).

Can't this fix (together with plugins update) fail e.g. when we require to test if user provided a _default_ value (i.e. empty password is a valid password)?

**Comment by TurboTurtle** [comment]
Closing this issue as this should be resolved with c71b415.
